src/Common/LRUFileCachePriority.h

```diff
@@ -0,0 +1,102 @@
+#pragma once
+
+#include <Common/IFileCachePriority.h>
+
+namespace DB
+{
+
+/// Based on the LRU algorithm implementation, the record with the lowest priority is stored at
+/// the head of the queue, and the record with the highest priority is stored at the tail.
+class LRUFileCachePriority : public IFileCachePriority
+{
+public:
+    using LRUQueue = std::list<FileCacheRecord>;
+    using LRUQueueIterator = typename LRUQueue::iterator;
+
+    class LRUFileCacheIterator : public IIterator
+    {
+    public:
+        LRUFileCacheIterator(LRUFileCachePriority * file_cache_, LRUQueueIterator queue_iter_)
+            : file_cache(file_cache_), queue_iter(queue_iter_)
+        {
+        }
+
+        void next() const override { queue_iter++; }
+
+        bool valid() const override { return (file_cache->queue.size() && (queue_iter != file_cache->queue.end())); }
```

Comment:
Looks like it should be enough to have
```suggestion
        bool valid() const override { return queue_iter != file_cache->queue.end()); }
```